src/item_factory.cpp

```diff
@@ -73,7 +73,8 @@ void Item_factory::finialize_item_blacklist()
         for( auto &elem : m_template_groups ) {
             elem.second->remove_item( itm );
         }
-        for( auto &recipes_b : recipes ) {
+        // FIXME: only recipes from by_category are erased;
+        for( auto &recipes_b : recipe_dict.by_category ) {
```

Comment:
Danger, Will Robinson! The function deletes the recipe (see few lines below). There *must* be no other place that stores that pointer as it becomes invalid.

You can probably solve this via a `remove` function in the `recipe_dict`, which should be called instead of the `delete`. The function should remove the pointer from all containers it might be in and finally call `delete`.